From bd48931bc142bdd104668f3a062a1f22600aae61 Mon Sep 17 00:00:00 2001
From: Paul Gortmaker <paul.gortmaker@windriver.com>
Date: Sun, 25 Jan 2009 17:58:09 -0500
Subject: [PATCH] modpost: mask trivial warnings

Newer HOSTCC will complain about various stdio fcns because
they are now declared with "warn_unused_return", or because
some initializers aren't filled out with values -- trivially
assign values so we aren't overwhelmed with these warnings
once the kernel build kicks off.

Signed-off-by: Paul Gortmaker <paul.gortmaker@windriver.com>

diff --git a/scripts/mod/modpost.c b/scripts/mod/modpost.c
index 48958d3cec9e..5911b2d18500 100644
--- a/scripts/mod/modpost.c
+++ b/scripts/mod/modpost.c
@@ -1929,7 +1929,7 @@ static void read_symbols(char *modname)
 	char *version;
 	char *license;
 	struct module *mod;
-	struct elf_info info = { };
+	struct elf_info info = { .hdr = NULL };
 	Elf_Sym *sym;
 
 	if (!parse_elf(&info, modname))
@@ -2355,7 +2355,7 @@ static int dump_sym(struct symbol *sym)
 
 static void write_dump(const char *fname)
 {
-	struct buffer buf = { };
+	struct buffer buf = { NULL, 0, 0 };
 	struct symbol *symbol;
 	int n;
 
@@ -2381,7 +2381,7 @@ struct ext_sym_list {
 int main(int argc, char **argv)
 {
 	struct module *mod;
-	struct buffer buf = { };
+	struct buffer buf = { NULL, 0, 0 };
 	char *kernel_read = NULL, *module_read = NULL;
 	char *dump_write = NULL, *files_source = NULL;
 	int opt;
-- 
2.10.1

